src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.WriteHelpers.cs

```diff
@@ -188,6 +189,132 @@ public partial class JsonTypeInfo<T>
             }
         }
 
+        internal async Task SerializeAsync(
+            PipeWriter utf8Json,
```

Comment:
Judging by how this method is implemented, am I right to think that this parameter could have been `IBufferWriter` instead? Could we consider making the APIs more general in this case? This would also match the API as exposed in `Utf8JsonWriter` as well:

https://learn.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonwriter.-ctor?view=net-8.0#system-text-json-utf8jsonwriter-ctor(system-buffers-ibufferwriter((system-byte))-system-text-json-jsonwriteroptions)

Reply:
Probably no since IBufferWriter doesn't inherently have a flushing API.